src/shim/ShimUtils.ts

```diff
@@ -111,19 +111,19 @@ export function validateDocumentData(data: Uint8Array) {
 }
 
 /**
- * Validate that the provided encrypted document is in the proper form, which should be Base64 encoded bytes of the document content and the AES IV prepended. Therefore
- * validate the length is at least 1 more than the IV length.
+ * Validate that the provided encrypted document is in the proper form, which should be Base64 encoded bytes or a Uint8Array of the document content and the AES IV prepended.
+ * Therefore validate the length is at least 1 more than the IV length.
  */
-export function validateEncryptedDocument(documentData: Base64Bytes) {
-    if (typeof documentData !== "string") {
-        throw new Error(`Invalid encrypted document format provided. Expected a string.`);
-    }
+export function validateEncryptedDocument(documentData: Base64Bytes | Uint8Array) {
     let encryptedByteLength: number;
+    if (typeof documentData !== "string" && documentData.constructor.name !== "Uint8Array") {
+        throw new Error("Invalid encrypted document content. Content should be a Base64 string or Uint8Array.");
+    }
     try {
-        //This will throw if the documentData isn't valid base64
-        encryptedByteLength = byteLength(documentData);
+        //The byteLength call will throw if the documentData isn't valid base64
+        encryptedByteLength = typeof documentData === "string" ? byteLength(documentData) : documentData.byteLength;
     } catch (_) {
-        throw new Error("Invalid encrypted document content. Content length is not a multiple of 4 which is required for base64 bytes.");
+        throw new Error("Invalid encrypted document content. Content length is not a multiple of 4.");
```

Comment:
The only way this catch can occur is the `byteLength` function so I don't think the error message should necessarily change. Personally I'd make this more clear by doing it all inside of an if/else:
```
if(isString){
    try{...}catch()
} else {
    documentData.byteLength
}
```

Reply:
Yeah, meant to change the comment back. I'm not sure what that change to doing it inside an if else clarifies. Keep in mind they aren't expressions in JS, so there'd be some more assignments in there.